flyteidl/protos/flyteidl/admin/execution.proto

```diff
@@ -430,3 +430,13 @@ message WorkflowExecutionGetMetricsResponse {
     // hierarchical structure using Flyte entity references.
     core.Span span = 1;
 }
+
+message ExecutionPhaseDeleteRequest {
```

Comment:
rather than requiring this deletion to be based on phase attributes only (what if one day we want to support timestamp filters instead?) can we make this more generic and re-use the ResourceListRequest (https://github.com/flyteorg/flyte/blob/a2331bd69b9f9ac38ce4f292c11d2374d93960db/flyteidl/protos/flyteidl/admin/common.proto#L203-L225) which includes out of the box support for filtering by phase, timestamp and other fields?

Reply:
something like


```
rpc DeleteExecutions (flyteidl.admin. ResourceListRequest) returns (flyteidl.admin.ExecutionDeleteResponse) {
		option (google.api.http) = {
      delete: "/api/v1/delete_execution/{id.project}/{id.domain}"
		};
		option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
		  description: "Delete all executions matching request filters"
		};
	};
}
```